sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/connection/MSSQLConnectionProvider.scala

```diff
@@ -35,10 +35,12 @@ private[sql] class MSSQLConnectionProvider extends SecureConnectionProvider {
     val configName = "jaasConfigurationName"
     val appEntryDefault = "SQLJDBCDriver"
 
+    val wrappedDriver = DriverRegistry.getWrappedDriver(driver)
```

Comment:
We cannot move this unwrap code in the caller side `ConnectionProvider.create`? If this issue can happen only in mssql and pg, the current code looks okay. But, I'm not sure about if that is true. IMO it seems safer to always unwrap drivers.

Reply:
Seems fine to me. Yeah is there any harm in always trying to unwrap?

Reply:
I think it's doable but have to take a deeper look at SPARK-6913 to be sure.

Reply:
After had a look it play a role only in class loading so we can pass the unwrapped `Driver` right from the beginning. I've changed the approach in the last commit. Thanks!